File: benchmark/corpus/javascript/sqli_safe_1.js
// SAFE: parameterized query with placeholder prevents SQL injection
const mysql = require('mysql2');
const express = require('express');
const app = express();

const db = mysql.createConnection({ host: 'localhost', user: 'root', database: 'app' });

app.get('/user', (req, res) => {
    const userId = req.query.id;
    const query = 'SELECT * FROM users WHERE id = ?';
    db.query(query, [userId], (err, results) => {
        if (err) return res.status(500).json({ error: err.message });
        res.json(results);
    });
});

app.listen(3000);

File: benchmark/corpus/python/sqli_safe_1.py
"""SAFE: parameterized query prevents SQL injection."""
import sqlite3


def find_user(username):
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    query = "SELECT * FROM users WHERE username = ?"
    cursor.execute(query, (username,))
    return cursor.fetchone()


def login(username, password):
    user = find_user(username)
    if user and user[2] == password:
        return True
    return False
